libbeat/processors/actions/decode_json_fields_test.go

```diff
@@ -199,6 +199,53 @@ func TestTargetRootOption(t *testing.T) {
 	assert.Equal(t, expected.String(), actual.String())
 }
 
+func TestArrayWithArraysDisabled(t *testing.T) {
```

Comment:
Do you mind adding test cases for decoding array when `process_array` is set to true? Also, it would be nice to see tests for `max_depth`. The coverage of `decode_json_fields` is pretty thin. :(

Reply:
Here it is: https://github.com/elastic/beats/pull/11318/files/193fb05581182455aa63ddadf8b998bfda8fb6c5#diff-baa231b1fdf8858ba8a286d1cc957e82R235